src/proc/y16i-to-y16y16.cpp

```diff
@@ -0,0 +1,39 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2022 Intel Corporation. All Rights Reserved.
+
+#include "y16i-to-y16y16.h"
+#include "stream.h"
+// CUDA TODO
+//#ifdef RS2_USE_CUDA
+//#include "cuda/cuda-conversion.cuh"
+//#endif
+
+namespace librealsense
+{
+    struct y16i_pixel { uint8_t rl : 8, rh : 4, ll : 8, lh : 4; int l() const { return lh << 8 | ll; } int r() const { return rh << 8 | rl; } };
+    void unpack_y16_y16_from_y16i_10(byte* const dest[], const byte* source, int width, int height, int actual_size)
+    {
+        auto count = width * height;
+// CUDA TODO
+//#ifdef RS2_USE_CUDA
+//        rscuda::split_frame_y16_y16_from_y12i_cuda(dest, count, reinterpret_cast<const y12i_pixel*>(source));
+//#else
+        split_frame(dest, count, reinterpret_cast<const y16i_pixel*>(source),
+            [](const y16i_pixel& p) -> uint16_t { return p.l() << 6 | p.l() >> 4; },  // We want to convert 10-bit data to 16-bit data
```

Comment:
I think it is not needed here - the 10 data is already unpacked and resides within the 16bit words.
So memcpy or direct cast each two bytes to a word can probably suffice.
Unless you're pushing the data to be found in MSB and not LSB as in the original stream

Reply:
changed to static_cast<uint16_t>